back/app/models/static_page.rb

```diff
@@ -20,30 +37,70 @@
 class StaticPage < ApplicationRecord
   CODES = %w[about terms-and-conditions privacy-policy faq proposals custom].freeze
 
+  has_many :pins, as: :page, inverse_of: :page, dependent: :destroy
+  has_many :pinned_admin_publications, through: :pins, source: :admin_publication
   has_one :nav_bar_item, dependent: :destroy
   has_many :static_page_files, -> { order(:ordering) }, dependent: :destroy
   has_many :text_images, as: :imageable, dependent: :destroy
   accepts_nested_attributes_for :nav_bar_item
   accepts_nested_attributes_for :text_images
 
+  before_validation :set_code, on: :create
+  before_validation :generate_slug, on: :create
+
+  before_validation :strip_title
+  before_validation :sanitize_top_info_section_multiloc
+  before_validation :sanitize_bottom_info_section_multiloc
+
+  before_destroy :confirm_is_custom, prepend: true
+
   validates :title_multiloc, presence: true, multiloc: { presence: true }
-  validates :body_multiloc, presence: true, multiloc: { presence: true, html: true }
   validates :slug, presence: true, uniqueness: true
   validates :code, inclusion: { in: CODES }
-  validates :code, uniqueness: true, if: ->(page) { !page.custom? }
+  validates :code, uniqueness: true, unless: :custom?
 
-  before_validation :set_code, on: :create
-  before_validation :generate_slug, on: :create
+  validates :banner_enabled, inclusion: [true, false]
+  validates :banner_layout, inclusion: %w[full_width_banner_layout two_column_layout two_row_layout]
+  validates :banner_overlay_color, css_color: true
```

Comment:
Does a custom page really have a banner overlay? I thought these are settings specific to the home page, but maybe I'm wrong in my understanding of what that feature does.

Reply:
It has a banner section on [the design](https://www.figma.com/file/XD9LDCrg2wnthCU8EEDTnL/Flexible-pages?node-id=1761%3A27228). We have discussed the different way that home page treats signed in / signed out settings compared to custom pages many times, and the shared assumption always seems to have been that we do have all these banner options for custom pages.

We could revisit that decision, of course.